processors/zipkin/src/main/java/org/hawkular/apm/processor/zipkin/SpanDeriverUtil.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.apm.processor.zipkin;
+
+import java.util.List;
+
+import org.hawkular.apm.server.api.model.zipkin.Span;
+
+/**
+ * Class for deriving various information from {@link Span}.
+ *
+ * @author Pavol Loffay
+ */
+public class SpanDeriverUtil {
+
+    private SpanDeriverUtil() {}
+
+    /**
+     * Derives http operation from span.
+     *
+     * @param span the Span
+     * @return operation (e.g. HTTP GET/POST)
+     */
+    public static String deriveHttpOperation(Span span) {
+        List<HttpCodesUtil.HttpCode> httpStatusCodes = HttpCodesUtil.getHttpStatusCodes(span.getBinaryAnnotations());
+        if (httpStatusCodes.size() > 0) {
+            return span.getName();
+        }
```

Comment:
If we determine it is a http span, then I think we should check that the name is one of the HTTP methods, e.g. GET, POST, etc. before returning it. If it isn't then we will have to have another strategy.